pkg/agent/controller/bgp/controller_test.go

```diff
@@ -2083,3 +2117,68 @@ func waitAndGetDummyEvent(t *testing.T, c *fakeController) {
 func doneDummyEvent(t *testing.T, c *fakeController) {
 	c.queue.Done(dummyKey)
 }
+
+func TestGetBGPPolicyInfo(t *testing.T) {
+	testCases := []struct {
+		name                  string
+		policiesToAdd         []runtime.Object
+		objects               []runtime.Object
+		existingState         *bgpPolicyState
+		expectedBgpPolicyName string
+		expectedASN           int32
+		expectedRouterID      string
+		expectedListenPort    int32
+	}{
+		{
+			name: "effective BGPPolicy exist",
+			policiesToAdd: []runtime.Object{generateBGPPolicy(bgpPolicyName1,
+				creationTimestamp,
+				nodeLabels1,
+				179,
+				65000,
+				true,
+				false,
+				true,
+				true,
+				false,
+				nil),
+			},
+			objects: []runtime.Object{
+				node,
+			},
+			existingState: generateBGPPolicyState(bgpPolicyName1,
+				179,
+				65000,
+				nodeAnnotations1[types.NodeBGPRouterIDAnnotationKey],
+				nil,
+				nil,
+			),
+			expectedBgpPolicyName: bgpPolicyName1,
+			expectedASN:           int32(65000),
+			expectedRouterID:      nodeAnnotations1[types.NodeBGPRouterIDAnnotationKey],
+			expectedListenPort:    int32(179),
+		},
+		{
+			name:          "effective BGPPolicy doesn't exist",
+			existingState: generateBGPPolicyState("", 0, 0, "", nil, nil),
```

Comment:
if the BGPPolicy didn't exist, then `c.bgpPolicyState` would be `nil`, so your test doesn't match